.github/workflows/create-condalock-file.yml

```diff
@@ -63,6 +63,8 @@ jobs:
         run: esmvaltool --help
       - shell: bash -l {0}
         run: esmvaltool version
+      - shell: bash -l {0}
+        run: flake8 -j 4
```

Comment:
You'll probably want to set the number of jobs on GitHub actions to 2 for Linux and 3 for osx: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

Reply:
does it matter? I got two cores too on me trusted Toshiba, but I can thread as many pytest processes as I want - well, as many threads as I have

Reply:
Yes, if you have more jobs running than CPU cores, the computer will start [context switching](https://en.wikipedia.org/wiki/Context_switch), which will slow things down. You should either not specify the number of jobs at all (flake8 tries to automatically detect the correct number) or set it to the number of CPU cores.

Reply:
OK I'll let flake8 decide but am not changing the pytest -n numbers, coz that's too much work :grin: Also is that thread context switching or process context switching coz if it's thread then we should be fine - 2 cores with at least two threads each should be enough for -n 4

Reply:
nevermind my blabber - we have `pytest -n 2` set everywhere and the OSX tests are slow just because of OSX not because they don't use all the 3 allocated cores, also 2 or 3 cores? What's this, 1998? :laughing:

Reply:
> OK I'll let flake8 decide but am not changing the pytest -n numbers, coz that's too much work

:+1: 

> Also is that thread context switching or process context switching coz if it's thread then we should be fine - 2 cores with at least two threads each should be enough for -n 4

If you think this is something worth spending time on, then I would recommend running some experiments with different settings

Reply:
nah man, see my latest comment - I thinks we're all good, Max Verstappen Red Bull performance :cow2: :netherlands: